gazebo_ros2_control/src/gazebo_system.cpp

```diff
@@ -0,0 +1,389 @@
+  this->dataPtr->joint_effort_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_pos_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_vel_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_eff_cmd_.resize(this->dataPtr->n_dof_);
+
+  gazebo::physics::PhysicsEnginePtr physics = gazebo::physics::get_world()->Physics();
+
+  std::string physics_type_ = physics->GetType();
+  if (physics_type_.empty()) {
+    RCLCPP_ERROR(this->nh_->get_logger(), "No physics engine configured in Gazebo.");
+    return false;
+  }
+
+  for (unsigned int j = 0; j < this->dataPtr->n_dof_; j++) {
+    //
+    // Perform some validation on the URDF joint and actuator spec
+    //
+    // Check that this transmission has one joint
+    if (transmissions[j].joints.empty()) {
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "Transmission " << transmissions[j].name <<
+          " has no associated joints.");
+      continue;
+    } else if (transmissions[j].joints.size() > 1) {
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "Transmission " << transmissions[j].name <<
+          " has more than one joint. Currently the default robot hardware simulation " <<
+          " interface only supports one.");
+      continue;
+    }
+    std::string joint_name = this->dataPtr->joint_names_[j] = transmissions[j].joints[0].name;
+
+    std::vector<std::string> joint_interfaces = transmissions[j].joints[0].interfaces;
+    if (joint_interfaces.empty() &&
+      !(transmissions[j].actuators.empty()) &&
+      !(transmissions[j].actuators[0].interfaces.empty()))
+    {
+      // TODO(anyone): Deprecate HW interface specification in actuators in ROS 2
+      joint_interfaces = transmissions[j].actuators[0].interfaces;
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "The <hardware_interface> element of transmission " <<
+          transmissions[j].name << " should be nested inside the <joint> element," <<
+          " not <actuator>. The transmission will be properly loaded, but please update " <<
+          "your robot model to remain compatible with future versions of the plugin.");
+    }
+    if (joint_interfaces.empty()) {
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "Joint " << transmissions[j].name <<
+          " of transmission " << transmissions[j].name <<
+          " does not specify any hardware interface. " <<
+          "Not adding it to the robot hardware simulation.");
+      continue;
+    } else if (joint_interfaces.size() > 1) {
+      // only a warning, allow joint to continue
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "Joint " << transmissions[j].name <<
+          " of transmission " << transmissions[j].name <<
+          " specifies multiple hardware interfaces. " <<
+          "Currently the default robot hardware simulation interface only supports one." <<
+          "Using the first entry");
+    }
+    std::string hardware_interface = joint_interfaces.front();
```

Comment:
this should be a parameter in the ros2_control tag and be parsed in the `configure` step accordingly.
c.f. https://github.com/ros-controls/ros2_control/blob/master/hardware_interface/include/hardware_interface/hardware_info.hpp